### DISCUSSION TITLE
How to include a PSScriptAnalyzer Custom Rule in this scenario?

### DISCUSSION THREAD

**Discussion by donk-msft** [oxsecurity/megalinter#2758] [Q&A] [answered]
I'd like to have a central repo with the mega-linter configuration that I can invoke when submitting a PR in Azure DevOps to other repos.  In Azure DevOps I've configured a branch policy with a build validation rule that invokes a minimal pipeline (in the repo to be checked) which in turn invokes the pipeline in the central repo.

This means that the folder structure on my build agent looks like this:
/home/vsts/work/1/s/MyCode
/home/vsts/work/1/s/MegaLinter

I have created a custom PSScriptAnalyzer rule that is stored in:
/home/vsts/work/1/s/MegaLinter/src/pssa/Measure-Case.psm1

The settings file is located in:
/home/vsts/work/1/s/MegaLinter/.powershell-psscriptanalyzer.psd1
Content:
```
# PSScriptAnalyzerSettings.psd1
@{
    CustomRulePath = @(
        '.\MegaLinter\src\pssa\Measure-Case.psm1'
    )

    IncludedRules = @(
        'Measure-*'
    )
}
```

I then added these environment variables:
-e POWERSHELL_POWERSHELL_CONFIG_FILE='.powershell-psscriptanalyzer.psd1' \
-e POWERSHELL_POWERSHELL_RULES_PATH='$(System.DefaultWorkingDirectory)/MegaLinter' \

Any thoughts why the custom rule is not included in the tests?

**Comment by nvuillam** [comment]
Please can you run MegaLinter with `LOG_LEVEL: DEBUG` so you can see the exact call to PS Script Analyzer ?

**Reply by donk-msft** [reply]
Thanks nvuillam. I set the log level and this is the relevant output (I think):

![image](https://github.com/oxsecurity/megalinter/assets/17045753/b95cb654-db04-4eda-8944-c0a402d0cae3)

**Reply by nvuillam** [reply]
Do you know the exact command line that should be run to fit your case ? (I'm not a PWSH expert ^^ )

**Comment by donk-msft** [comment]
I'd expect this command to be run in the scenario where I want the PSScriptAnalyzer default rules and my custom rules to be validated:

Invoke-ScriptAnalyzer -Path .\MyScript.ps1 -CustomRulePath .\MyRule.psm1 -IncludeDefaultRules

**Comment by donk-msft** [comment, marked as answer]
In the meantime I've revised the setup and it's working now. In my original question, there's an error in the rules file. 'IncludedRules' should be 'IncludeRules'. 

I have my rules files in the root of my repo and that does the trick for me.
